pkg/client/unversioned/clientcmd/auth_loaders.go

```diff
@@ -27,6 +27,9 @@ import (
 	clientauth "k8s.io/kubernetes/pkg/client/unversioned/auth"
 )
 
+// promptAuthCache is used to stone prompt auth information
+var promptAuthCache = make(map[string]string)
```

Comment:
No globals, but I understand what you're trying to do now.  How about two separate fields on the `PromptingAuthLoader`, same file.